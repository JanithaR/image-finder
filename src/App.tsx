/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  StatusBar,
  TextInput,
  Pressable,
  View,
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

import strings from './strings';
import colors from './colors';

declare const global: { HermesInternal: null | {} };

const App = () => {
  function onSearchPress() {
    console.log('on search press');
  }

  return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView style={styles.mainWrapper}>
        <View style={styles.searchWrapper}>
          <TextInput
            placeholder={strings.searchImages}
            style={styles.searchInput}
          />
          <Pressable
            onPress={onSearchPress}
            style={({ pressed }) => [
              {
                backgroundColor: pressed
                  ? colors.primaryGreenPressed
                  : colors.primaryGreen,
              },
              styles.searchButton,
            ]}
            accessibilityRole="button">
            <Icon name="image-search" size={30} color={colors.primaryYellow} />
          </Pressable>
        </View>
      </SafeAreaView>
    </>
  );
};

const searchBorderRadius: number = 10;

const styles = StyleSheet.create({
  mainWrapper: {
    padding: 20,
    flex: 1,
  },
  searchInput: {
    borderWidth: 1,
    borderColor: colors.searchInputBorder,
    borderTopLeftRadius: searchBorderRadius,
    borderBottomLeftRadius: searchBorderRadius,
    paddingHorizontal: 15,
    fontSize: 18,
    flex: 1,
  },
  searchButton: {
    borderTopRightRadius: searchBorderRadius,
    borderBottomRightRadius: searchBorderRadius,
    width: 60,
    justifyContent: 'center',
    alignItems: 'center',
  },
  searchWrapper: {
    flexDirection: 'row',
  },
});

export default App;
