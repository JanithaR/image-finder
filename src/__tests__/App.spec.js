import React from 'react';
import { render } from '@testing-library/react-native';

import App from 'src/App';

function setup() {
  return render(<App />);
}

describe('Search', () => {
  it('should render styles correctly', () => {
    const { toJSON } = setup();

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render a search input', () => {
    const { queryByPlaceholderText } = setup();

    expect(queryByPlaceholderText('Search Images')).toBeTruthy();
  });

  it('should render a search button', () => {
    const { queryByA11yRole } = setup();

    expect(queryByA11yRole('button')).toBeTruthy();
  });
});
