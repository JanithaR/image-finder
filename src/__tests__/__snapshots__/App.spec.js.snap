// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render styles correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Search Images"
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomLeftRadius": 10,
          "borderColor": "#ddd",
          "borderTopLeftRadius": 10,
          "borderWidth": 1,
          "flex": 1,
          "fontSize": 18,
          "paddingHorizontal": 15,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#489c91",
          },
          Object {
            "alignItems": "center",
            "borderBottomRightRadius": 10,
            "borderTopRightRadius": 10,
            "justifyContent": "center",
            "width": 60,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#f6c446",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°¥·
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
